include/Flow.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <queue>
+
+class Flow {
+public:
+    explicit Flow(std::shared_ptr<NetworkDevice> sender, std::shared_ptr<NetworkDevice> receiver,
+                  std::uint64_t packet_generation_interval_ns);
+
+    void Send();
+
+    std::shared_ptr<NetworkDevice> m_sender;
```

Comment:
Should these members be *private*?

Reply:
Yes, I think every member, except constructor, should be private

Reply:
Yes, I think every member, except constructor, should be private